fastjet/internal/delaunay/point_test.go

```diff
@@ -77,6 +78,34 @@ func TestFindNearest(t *testing.T) {
 	}
 }
 
+func TestFindSecondNearest(t *testing.T) {
+	const tol float64 = 0.001
+	tests := []struct {
+		p                 *Point
+		adjacentTriangles []*Triangle
+		wantDist          float64
+		wantNeighbor      *Point
+	}{
+		{
```

Comment:
ok.

could you also add:

- a test that triggers the `if p.dist2 == 0 {` of line 71?
I'd like to see how this plays out later on wrt the (now `if-else-if`, later) `switch` which should always take the `dist < secondMin` branch...
- a test that triggers the `if p2.dist2 == 0` of line 96
- and the same for `p3` ?

Reply:
Done

In the test function:
The first test is without duplicates
The second test will trigger `if p.dist2 == 0{`
The third test will trigger `if p2.dist2 == 0{`
And the fourth test will trigger `if p3.dist2 == 0{`